docs/synapse/devguides/power-ups.rst

```diff
@@ -0,0 +1,426 @@
+          - type: str
+            help: A mandatory / positional command line argument.
+
+A more complete example of help output::
+
+    storm> acme.hello.omgopts --help
+
+    This is a mult-line description containing usage examples.
+
+    // run the command with some nodes
+    inet:fqdn=acme.newp | acme.hello.omgopts vertex.link
+
+    // run the command with some command line switches
+    acme.hello.omgopts --debug --hehe haha vertex.link
+
+
+    Usage: acme.hello.omgopts [options] <fqdn>
+
+    Options:
+
+      --help                      : Display the command usage.
+      --hehe <hehe>               : The value of the hehe optional input.
+      --debug                     : Enable debug output.
+
+    Arguments:
+
+      <fqdn>                      : A mandatory / positional command line argument.
+    complete. 0 nodes in 6 ms (0/sec).
+
+Command line options are available within the **Storm** command by accessing the implicit
+``$cmdopts`` variable.
+
+``storm/commands/acme.hello.omgopts.storm``::
+
+    // An init {} block only runs once even if there are multiple nodes in the pipeline.
+    init {
+
+        // set global debug (once) if the user specified --debug
+        if $cmdopts.debug { $lib.debug = $lib.true } }
+
+        if ($cmdopts.hehe) { $lib.print("User Specified hehe: {hehe}", hehe=$cmdopts.hehe) }
+
+        // Normalize the FQDN in case we want to send it to an external system
+        ($ok, $fqdn) = $lib.trycast($cmdopts.fqdn, inet:fqdn)
+          
+        if (not $ok) {
+            $lib.exit("Invalid FQDN Specified: {fqdn}", ipv4=$cmdopts.fqdn)
+        }
+
+        // Maybe call an API here or something...
+        $lib.print("FQDN: {fqdn}", fqdn=$fqdn)
+    }
+
+
+    // You may also act on nodes in the pipeline
+    $lib.print("GOT NODE: {repr}", repr=$node.repr())
+
+    // Any nodes in the pipeline here are yielded to the caller
+
+Command Option Conventions
+--------------------------
+
+--help
+  This option is reserved and handled automatically to print a command usage statement which also enumerates any
+  positional or optional arguments.
+
+--debug
+  This option is typically used to enable debug output in the **Storm** interpreter by setting the ``$lib.debug``
+  variable if it us specified. The ``$lib.debug`` variable has a recursive effect and will subsequently enable
+  debug output in any command or functions called from the command.
+
+--yield
+  By default, a command is generally expected to yield the nodes that it recieved as input from the pipeline. In
+  some instances it is useful to instruct the command to yield the nodes it creates. For example, if you specify
+  ``inet:fqdn`` nodes as input to a DNS resolver command, it may be useful to tell the command to yield the newly
+  created ``inet:dns:a`` records rather than the input ``inet:fqdn`` nodes.  Commands frequently use the ``divert``
+  **Storm** command to implement ``--yield`` functionality.
+
+--asof <time>
```

Comment:
Might be good to show or describe the convention of namespacing the cache keys

Reply:
I think i'll add that to advanced topics :)